tests/disaster-recovery/mdr/test_app_failover_and_relocate_when_zone_down.py

```diff
@@ -0,0 +1,272 @@
+
+        # Restore new hub
+        restore_backup()
+
+        # Verify the managed clusters are imported in new hub
+        for cluster in get_non_acm_cluster_config():
+            validate_cluster_import(cluster.ENV_DATA["cluster_name"])
+
+        # Wait or verify the drpolicy is in validated state
+        if config.RUN.get("mdr_failover_via_ui"):
+            verify_drpolicy_ui(acm_obj, 0)
+        else:
+            verify_drpolicy_cli()
+
+        # Verify if cluster is marked unavailable on ACM console
+        if config.RUN.get("mdr_failover_via_ui"):
+            config.switch_acm_ctx()
+            check_cluster_status_on_acm_console(
+                acm_obj,
+                down_cluster_name=self.primary_cluster_name,
+                expected_text="Unknown",
+            )
+
+        # ToDo: Deploy application in both managed cluster and
+        #  to verify the applications are present in secondary cluster
+
+        # Fenced the primary managed cluster
+        enable_fence(drcluster_name=self.primary_cluster_name)
+
+        # Application Failover to Secondary managed cluster
+        secondary_cluster_name = get_current_secondary_cluster_name(
+            workload.workload_namespace
+        )
+        if config.RUN.get("mdr_failover_via_ui"):
+            logger.info("Start the process of Failover from ACM UI")
+            config.switch_acm_ctx()
+            failover_relocate_ui(
+                acm_obj,
+                workload_to_move=f"{workload.workload_name}-1",
+                policy_name=workload.dr_policy_name,
+                failover_or_preferred_cluster=secondary_cluster_name,
+            )
+        else:
+            failover(
+                failover_cluster=secondary_cluster_name,
+                namespace=workload.workload_namespace,
+            )
+
+        # Verify application are running in other managedcluster
+        # And not in previous cluster
+        set_current_primary_cluster_context(workload.workload_namespace)
+        wait_for_all_resources_creation(
+            workload.workload_pvc_count,
+            workload.workload_pod_count,
+            workload.workload_namespace,
+        )
+
+        # Verify the failover status from UI
+        if config.RUN.get("mdr_failover_via_ui"):
+            config.switch_acm_ctx()
+            verify_failover_relocate_status_ui(acm_obj)
+
+        # Start nodes of the managed cluster which is down
+        wait_time = 120
+        logger.info(
+            f"Wait for {wait_time} seconds before starting the nodes of managed cluster which is down"
+        )
+        time.sleep(wait_time)
+        nodes_multicluster[managed_cluster_index].start_nodes(managed_cluster_node_objs)
+        logger.info(
+            f"Waiting for {wait_time} seconds after starting nodes of previous primary cluster"
+        )
+        time.sleep(wait_time)
+        wait_for_nodes_status([node.name for node in managed_cluster_node_objs])
```

Comment:
You can use "wait_for_cluster_connectivity()" routine to check cluster is up and running.

Reply:
added